src/core/src/pass/pass.cpp

```diff
@@ -72,21 +72,9 @@ class RunLocker {
 
 ov::pass::ModelPass::~ModelPass() = default;
 
-OPENVINO_SUPPRESS_DEPRECATED_START
-
 bool ov::pass::ModelPass::run_on_model(const std::shared_ptr<ov::Model>& m) {
     RUN_ON_MODEL_SCOPE(ModelPass);
     RunLocker locked(call_on_model);
-    OPENVINO_ASSERT(!call_on_function,
-                    "Cycle detected. run_on_model() or run_on_function() method should be overridden.");
-    bool sts = run_on_function(m);
```

Comment:
@ilya-lavrenov @ilyachur Pls pay more attention here, I'm not sure the change here is right!

Reply:
It is not right. We need to remove the default implementation in this case.

Reply:
Remove the implementation and make a pure virtual func.